src/app/api/stories/page_views/route.ts

```diff
@@ -20,7 +20,49 @@ export async function GET(req: Request) {
       },
     });
 
-    return NextResponse.json({ count });
+    return NextResponse.json({ total: count || 0 });
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientValidationError) {
+      console.log(e.message);
+      return NextResponse.json({ error: "Bad Request" }, { status: 400 });
+    }
+
+    // all other errors
+    return NextResponse.json({ error: e }, { status: 500 });
+  }
+}
+export async function POST(req: Request) {
+  try {
+    const parsedParams = postSchema.safeParse(await req.json());
+    if (!parsedParams.success) {
+      return NextResponse.json(parsedParams.error, { status: 400 });
+    }
+    const { start_date, end_date, topic } = parsedParams.data;
+    const countPromise = prisma.pageView.count({
+      where: {
+        story: {
+          topics: {
+            has: topic,
+          },
+        },
+        createdAt: {
+          lte: new Date(end_date),
+          gte: new Date(start_date),
+        },
+      },
+    });
+    const totalPromise = prisma.pageView.count({
+      where: {
+        createdAt: {
+          lte: new Date(end_date),
+          gte: new Date(start_date),
+        },
+      },
+    });
+    const [count, total] = await Promise.all([countPromise, totalPromise]);
+    return NextResponse.json({
+      avg: total && count ? Math.round((count / total) * 10000) / 100 : 0,
```

Comment:
In this case, it is calculating the percentage that show amount of people reading specific given topic in given time.
The formula is (total view of specific topic) divided by (total view of all topic), and then multiply by 100. The formula apply to the code is little bit diffence, it is because it should round up to 2 decimal places. I do it without the rounding if you want.